crates/iceberg/src/spec/schema.rs

```diff
@@ -137,9 +142,55 @@ impl SchemaBuilder {
             name_to_id,
             lowercase_name_to_id,
             id_to_name,
+
+            field_id_to_accessor,
         })
     }
 
+    fn build_accessors(&self) -> HashMap<i32, Arc<StructAccessor>> {
```

Comment:
How about using SchemaVisitor to build this?

Reply:
Also it would be better to add some uts to show the result.

Reply:
Will do, re SchemaVisitor. Sorry Renjie, not sure what you mean with the second comment though?

Reply:
`SchemaVisitor` won't work for this without some modifications. We need to know the index within `Schema.fields` for each field to create its accessor, and `visit_struct` does not provide this at the moment.

`visit_struct` and `SchemaVisitor` could be modified to pass the field index to the visitor's methods. Can we keep this PR as-is and have a TODO so that we can move on with this working as-is and I or someone else can pick up this refactor in another PR?

Reply:
> Will do, re SchemaVisitor. Sorry Renjie, not sure what you mean with the second comment though?

Sorry, I mean adding some unit test and example code in api to demonstrate its usage, like what python does here: https://github.com/apache/iceberg-python/blob/d3db8401cd006fcbacf1d9a2502248ce4228ff06/pyiceberg/schema.py#L1162

Reply:
> SchemaVisitor won't work for this without some modifications. We need to know the index within Schema.fields for each field to create its accessor, and visit_struct does not provide this at the moment.

Sorry, I don't get your point, following code would work?
```rust
fn visit_struct(&self, struct: &Struct, field_results: Vec<Option<StructAccessor>>) {
  struct.fields.zip(field_result).enumerate().map(|(index, (field, field_result)|) {}
}
```

Reply:
Sorry Renjie, I still don't follow! `visit_struct` already exists and has a different signature to that? 

https://github.com/apache/iceberg-rust/blob/4e89ac71c3ea77b9270b71dac52b5c3e50c36526/crates/iceberg/src/spec/schema.rs#L365

Reply:
Sorry for misclarification, in fact I mean the `r#struct` method in `SchemaVisitor`: https://github.com/apache/iceberg-rust/blob/1c2a20b13e67e4f91a44d49fa1b0e3432cd48432/crates/iceberg/src/spec/schema.rs#L324

But I agree that it could be left to do it later.